EdgeReachTech/adgereachtech-web-bn
# Message

What does this PR do?

This PR creates the message model, controller, and router for handling messages between users in a chat application.

Description of Task to be completed?

Implement the message model with Mongoose.
Set up the message controller with methods for creating, updating, reading, and deleting messages.
Define the message routes for the above operations.

How should this be manually tested?

Create a Message:

Endpoint: POST /message/createMessage/:receiverId
Payload: { "content": "Hello, how are you?" }
Ensure a valid JWT token is provided in the request headers.

Update a Message:

Endpoint: PATCH /message/updateMessage/:messageId
Payload: { "content": "Updated message content" }
Ensure a valid JWT token is provided in the request headers.

Read a Message:

Endpoint: PATCH /message/readMessage/:messageId
Ensure a valid JWT token is provided in the request headers.

Delete a Message:

Endpoint: DELETE /message/deleteMessage/:messageId
Ensure a valid JWT token is provided in the request headers.
